lib/utils/reader.cpp

```diff
@@ -7,79 +7,101 @@ namespace utils{
 
 Reader::Reader(){}
 
-//this function is probably now useless but still good practise!
-void Reader::readFirstProblem(const std::string problems_filepath){
+void Reader::readNextProblem(std::fstream &problems_filestream){
+  //clear anything still sored in class members from previous read
+  problem_matrix_.clear();
+  lower_bounds_.clear();
+  upper_bounds_.clear();
 
-  //initialise filestream and open file
-  std::fstream newfile;
-  newfile.open(problems_filepath, std::ios::in);
   //check file is open
-  if(!newfile.is_open()){
+  if(!problems_filestream.is_open()){
     std::cout << "ERROR: Unable to open file" << std::endl;
   }
+  
   //initialise strings for storing elements read from file
   std::string temp_string;
   std::string num_variables_string;
   std::string num_inequality_rows_string;
+  std::string num_equality_rows_string;
 
-  // get number of variables and number of inequalities we are reading in
-  std::getline(newfile, num_variables_string);
-  const core::int_t num_variables = static_cast<core::int_t>(atoi(num_variables_string.c_str())) + 1;
+  // get number of variables we are reading in, taking into 
+  // account whether the line is a separation line or not 
+  // (it is only not a separation line in the first problem.)
+  std::getline(problems_filestream, temp_string);
+  if (temp_string.find('~') != std::string::npos) {
+    std::getline(problems_filestream, num_variables_string);
+  } else {
+    num_variables_string = temp_string;
+  }
+  const int num_variables = static_cast<int>(atoi(num_variables_string.c_str()));
 
-  std::getline(newfile, num_inequality_rows_string);
-  const core::int_t num_inequality_rows = static_cast<core::int_t>(atoi(num_inequality_rows_string.c_str()));
+  // Get number of inequalities.
+  std::getline(problems_filestream, num_inequality_rows_string);
+  const int num_inequality_rows = static_cast<int>(atoi(num_inequality_rows_string.c_str()));
 
-  //initialise vector for using as temporary holding for matrix row
-  std::vector<core::int_t> matrix_row;
+  // Define vectors to read in constraints.
+  std::vector<int> problem_row; // A line in the input file
+  int constant_term;
```

Comment:
what does this do? naming isn't clear? Also a bit misleading since constant implies unchanging but this variable is reassigned a few times?